roles/translator/src/proxy/bridge.rs

```diff
@@ -141,57 +144,69 @@ impl Bridge {
     /// Receives a SV1 `mining.submit` message from the `Downstream`, translates it to a SV2
     /// `SubmitSharesExtended` message, and sends it to the `Upstream`.
     fn handle_downstream_share_submission(self_: Arc<Mutex<Self>>) {
-        let rx_sv1_submit = self_.safe_lock(|s| s.rx_sv1_submit.clone()).unwrap();
-        let tx_sv2_submit_shares_ext = self_
-            .safe_lock(|s| s.tx_sv2_submit_shares_ext.clone())
+        let (rx_sv1_submit, tx_sv2_submit_shares_ext, target, tx_status) = self_
+            .safe_lock(|s| {
+                (
+                    s.rx_sv1_submit.clone(),
+                    s.tx_sv2_submit_shares_ext.clone(),
+                    s.target.clone(),
+                    s.tx_status.clone(),
+                )
+            })
             .unwrap();
-        let target = self_.safe_lock(|s| s.target.clone()).unwrap();
-        let tx_status = self_.safe_lock(|s| s.tx_status.clone()).unwrap();
         task::spawn(async move {
             loop {
-                let target = target.safe_lock(|t| t.clone()).unwrap().try_into();
+                let target = target.safe_lock(|t| t.clone()).map_err(|_| PoisonLock);
+                let target = handle_result!(tx_status, target).try_into();
                 let upstream_target: [u8; 32] = handle_result!(tx_status, target);
                 let mut upstream_target: Target = upstream_target.into();
-                self_
+                let res = self_
                     .safe_lock(|s| s.channel_factory.set_target(&mut upstream_target))
-                    .unwrap();
+                    .map_err(|_| PoisonLock);
+                handle_result!(tx_status, res);
 
-                let (sv1_submit, extrnonce) = rx_sv1_submit.clone().recv().await.unwrap();
-                let channel_sequence_id =
-                    self_.safe_lock(|s| s.channel_sequence_id.next()).unwrap() - 1;
+                let (sv1_submit, extrnonce) =
+                    handle_result!(tx_status, rx_sv1_submit.clone().recv().await);
+                let channel_sequence_id = self_
+                    .safe_lock(|s| s.channel_sequence_id.next())
+                    .map_err(|_| PoisonLock);
+                let channel_sequence_id = handle_result!(tx_status, channel_sequence_id) - 1;
                 let sv2_submit = self_
                     .safe_lock(|s| s.translate_submit(channel_sequence_id, sv1_submit, extrnonce))
-                    .unwrap();
+                    .map_err(|_| PoisonLock);
+                let sv2_submit = handle_result!(tx_status, sv2_submit);
```

Comment:
Looks like a duplicate line

Reply:
`translate submit()` returns a result so it has to handle both the lock result and the `translate_submit()` result. Maybe nesting the `handle_result!()` will look cleaner

Reply:
I think if you leave a comment here that would be fine. Someone looking through this code casually might just think it's a duplicate line like i did and remove it.